discord-scripts/thread-management/auto-join.ts

```diff
@@ -84,10 +84,14 @@ async function autoJoinThread(
     )
     .reverse()
 
+  const normalize = (str: string) =>
+    str.toLowerCase().replace(/\s+/g, " ").trim()
```

Comment:
It feels like the space collapsing isn't really that necessary and we could just make sure we `trim` everywhere?

Separately though, we really need `normalize` to have a clearer name, and make it a top-level named `function`.